src/backend/distributed/utils/attribute.c

```diff
@@ -619,3 +604,126 @@ MultiTenantMonitorshmemSize(void)
 
 	return size;
 }
+
+
+/*
+ * ExtractTopComment extracts the top-level multi-line comment from a given input string.
+ */
+static char *
+ExtractTopComment(const char *inputString)
+{
+	int commentStartCharsLength = 2;
+	int inputStringLen = strlen(inputString);
+	if (inputStringLen < commentStartCharsLength)
+	{
+		return NULL;
+	}
+
+	int commentEndCharsIndex = 0;
+
+	/* If query doesn't start with a comment, return NULL */
+	if (inputString[commentEndCharsIndex] != '/' ||
+		inputString[commentEndCharsIndex + 1] != '*')
+	{
+		return NULL;
+	}
+
+	/* Skip the comment start characters */
+	commentEndCharsIndex += commentStartCharsLength;
+	while (commentEndCharsIndex < inputStringLen &&
+		   !(inputString[commentEndCharsIndex] == '*' &&
+			 inputString [commentEndCharsIndex + 1] == '/'))
+	{
+		commentEndCharsIndex++;
+	}
+
+	/* If there is no end of comment chars , return NULL */
+	if (inputString[commentEndCharsIndex] != '*' &&
+		inputString[commentEndCharsIndex + 1] != '/')
+	{
+		return NULL;
+	}
```

Comment:
This basically reimplements the C `strstr` function afaict. Let's use that one instead.